### PR TITLE
3DC multi branches cleanup

### PR DESCRIPTION
**Pull request by yux0**
In 3+DC, the mutable state may have multi-branches info. When a workflow is cleanup, all branches need to be deleted. Currently, the logic is to remove all non current branches (not max version), this is due to archival only archive current branch. In the future, when archival supports archive multiple branches. The cleanup can delete all branches.

### CODE REVIEW

**Comment by dmetzgar** [comment]
Fixes #1757

File: service/history/timerQueueProcessorBase.go [hunk 1]
```diff
@@ -704,15 +705,20 @@ func (t *timerQueueProcessorBase) deleteWorkflowHistory(task *persistence.TimerT
 				tag.TaskType(task.GetTaskType()))
 			versionHistories := msBuilder.GetAllBranches()
 
-			for _, versionHistory := range versionHistories.Histories {
+		CleanupLoop:
+			for idx, versionHistory := range versionHistories.Histories {
+				if reflect.DeepEqual(versionHistory.BranchToken, msBuilder.GetCurrentBranch()) {
```
Comment (wxing1292) [inline service/history/timerQueueProcessorBase.go]: bytes.Equal

File: service/history/timerQueueProcessorBase.go [hunk 2]
```diff
@@ -704,15 +705,20 @@ func (t *timerQueueProcessorBase) deleteWorkflowHistory(task *persistence.TimerT
 				tag.TaskType(task.GetTaskType()))
 			versionHistories := msBuilder.GetAllBranches()
 
-			for _, versionHistory := range versionHistories.Histories {
+		CleanupLoop:
+			for idx, versionHistory := range versionHistories.Histories {
```
Comment (wxing1292) [inline service/history/timerQueueProcessorBase.go]: probably also need to do a null check here

File: service/history/timerQueueProcessorBase.go (hunk 1 above)
Comment (yux0) [reply service/history/timerQueueProcessorBase.go]: Done

File: service/history/timerQueueProcessorBase.go (hunk 2 above)
Comment (yux0) [reply service/history/timerQueueProcessorBase.go]: Done

**Pull request merged**